torch/distributed/fsdp/flat_param.py

```diff
@@ -310,7 +311,7 @@ def _init_flat_param(
                         )
                     )
                 else:
-                    if isinstance(param, FlatParameter):
+                    if type(param) is FlatParameter:
```

Comment:
does this mean that previously, any Parameter subclass with _is_param == True would mistakenly pass this check as a FlatParameter?

Reply:
Yes.